src/ci/docker/host-x86_64/x86_64-gnu-tools/Dockerfile

```diff
@@ -23,7 +24,17 @@ RUN /scripts/cmake.sh
 
 COPY host-x86_64/x86_64-gnu-tools/checktools.sh /tmp/
 
+RUN curl -sL https://nodejs.org/dist/v14.4.0/node-v14.4.0-linux-x64.tar.xz | tar -xJ
+ENV PATH="/node-v14.4.0-linux-x64/bin:${PATH}"
+
+# Install required dependencies from browser-UI-test framework
+# For now, we need to use `--unsafe-perm=true` to go around an issue when npm tries
+# to create a new folder. For reference:
+# https://github.com/puppeteer/puppeteer/issues/375
+
 ENV RUST_CONFIGURE_ARGS \
   --build=x86_64-unknown-linux-gnu \
   --save-toolstates=/tmp/toolstate/toolstates.json
-ENV SCRIPT /tmp/checktools.sh ../x.py
+ENV SCRIPT /tmp/checktools.sh ../x.py && \
+  npm install browser-ui-test -g --unsafe-perm=true && \
+  cd .. && python3 ./x.py test src/test/rustdoc-gui --stage 2
```

Comment:
This is calling x.py from the current directory, but should be using ../x.py to create the build dir in the right place. We need to support invoking x.py from subdirectories, so paths should always be join'd with builder.config.src or something like that (which points to the root of the rust-lang/rust checkout).

Reply:
Makes sense. It originally was using `../` (without the the call to `cd`) but it couldn't find `stc/test/rustdoc-gui` so in both cases, it seems like a needed update. I'll send a PR to fix that tomorrow.